### PR TITLE
Update wdio to v7

### PR DESCRIPTION
**Pull request by pasierb**
*Description of changes:*

- Update wdio dependencies to v7
- change version to `7.0.0` to follow wdio dependencies versioning scheme

*Testing*

- tested locally

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/launcher.ts [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
   expiresInSeconds?: number;
 }
 
-export default class DeviceFarmLauncher implements WebdriverIO.HookFunctions {
+export default class DeviceFarmLauncher implements Services.ServiceInstance {
   private readonly devicefarm: DeviceFarm;
-  private readonly options: DeviceFarmConfig;
 
-  constructor(options: DeviceFarmConfig) {
+  constructor(private readonly _options: DeviceFarmConfig) {
```
Comment (pasierb) [inline src/launcher.ts:18]: Renamed `options` to `_options` due to conflict with `ServiceInstance` interface. It already has `options` defined as public member.

File: src/launcher.ts [hunk 2]
```diff
@@ -1,46 +1,46 @@
...
-      log.info("Created device farm test grid:", testGridUrlResult);
-
-      Object.assign(cap, {
-        protocol: "https",
-        port: 443,
-        hostname: url.hostname,
-        path: url.pathname,
-        connectionRetryTimeout: 180000,
-      });
+    if (Array.isArray(capabilities)) {
```
Comment (pasierb) [inline src/launcher.ts:28]: Need to check for array as `RemoteCapabilities` can be of type:

```
export interface MultiRemoteCapabilities {
    [instanceName: string]: WebDriverIOOptions;
}
```

Support for multiremote should be addressed separately.

File: src/launcher.ts (hunk 1 above)
Comment (JavierLuna) [reply src/launcher.ts:18]: 🦄 I think `serviceOptions` is a bit more descriptive than `_options`. I know it is internal to the library, but could be useful if someone looks at the constructor signature.

Comment (pasierb) [reply src/launcher.ts:18]: Not sure about that, but regardless of that I was trying to keep this PR as small as possible and focus only on v7 update. Consider renaming variables out of scope if possible.
It also follows the pattern from similar services like https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-browserstack-service/src/launcher.ts#L21.

Comment (JavierLuna) [reply src/launcher.ts:18]: 👍 Gotcha, let's keep it like it is then

**Review by JavierLuna — APPROVED** [review]
LGTM

**Review by caglaburcu — APPROVED** [review]

**Pull request merged**